publisher/sns/sns.go

```diff
@@ -67,18 +69,22 @@ func (p *Publisher) Publish(ctx context.Context, msg interface{}) error {
 // kept under 100 messages so that all messages can be published in 10 tries. In case
 // of failure when parsing or publishing any of the messages, this function will stop
 // further publishing and return an error
-func (p *Publisher) PublishBatch(ctx context.Context, msgs []interface{}) error {
+func (p *Publisher) PublishBatch(ctx context.Context, msgs []models.Message) (map[string]error, int64, int64, error) {
 	var (
 		defaultMessageGroupID = "default"
+		publishResult         = make(map[string]error)
 		err                   error
+
+		errorCount   int64
+		successCount int64
 	)
 
 	isFifo := strings.Contains(strings.ToLower(p.cfg.TopicArn), "fifo")
 
 	var (
 		numPublishedMessages = 0
 		start                = 0
-		end                  = 10 // 10 is the maximum batch size for SNS.PublishBatch
+		end                  = constants.MaxBatchSize
```

Comment:
why can't these be merged with variables on line 73-79?

Reply:
Done